### PR TITLE
support key rotation in terraform

### LINKED ISSUE #38: Create script that rotates the keys
Create a script that rotates the keys, by changing the key in key vault, then updating the key in the function app, then resetting the APIM cache.

### PR DESCRIPTION
**Pull request by linkinchow**
Close #38 
Will create a new issue for write up.

### CODE REVIEW

File: env/main.tf [hunk 1]
```diff
@@ -357,6 +357,14 @@ resource "azurerm_api_management_api_policy" "patient_policy" {
       <value>@((string)context.Variables["funchostkey"])</value>
     </set-header>
   </inbound>
+  <outbound>
+    <base />
+    <choose>
+      <when condition="@(context.Response.StatusCode == 401)">
+        <cache-remove-value key="func-host-key" />
```
Comment (hannesne) [inline env/main.tf:364]: Does this require the request to be resent, to get a new key?

Comment (linkinchow) [reply env/main.tf:364]: Yes, or I can just update the cache here.

Comment (linkinchow) [reply env/main.tf]: now it will update the host key when 401.

File: env/main.tf [hunk 2]
```diff
@@ -337,20 +337,20 @@ resource "azurerm_api_management_api_policy" "patient_policy" {
   <inbound>
     <base />
     <!-- Look for func-host-key in the cache -->
-    <cache-lookup-value key="func-host-key" variable-name="funchostkey" />
+    <cache-lookup-value key="${data.azurerm_key_vault_secret.fa_patient_api_host_key.version}" variable-name="funchostkey" />
```
Comment (hannesne) [inline env/main.tf:340]: we should probably prefix it with "func-host-key-" to prevent clashes.

Comment (linkinchow) [reply env/main.tf:340]: will fix it soon

**Pull request merged**